tokio/src/runtime/task/waker.rs

```diff
@@ -15,7 +15,7 @@ pub(super) struct WakerRef<'a, S: 'static> {
 
 /// Returns a `WakerRef` which avoids having to pre-emptively increase the
 /// refcount if there is no need to do so.
-pub(super) fn waker_ref<T, S>(header: &Header) -> WakerRef<'_, S>
+pub(super) fn waker_ref<T, S>(header: &NonNull<Header>) -> WakerRef<'_, S>
```

Comment:
`NonNull` is `Copy`, so I don't totally understand why this can't just take the `NonNull` by value, and not dereference it when calling `raw_waker`? previously this code took a reference to a `Header`, but now it takes a reference to a pointer to a header...

if there's a reason we need to take `&NonNull<Header>`, it would be nice to maybe have a comment explaining that.

Reply:
Well, if we removed the reference, then the lifetime on `WakerRef` is no longer bound to anything. That's the only reason.

Reply:
oh, yeah, i see that now. never mind!